satellite/metainfo/batch.go

```diff
@@ -0,0 +1,215 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package metainfo
+
+import (
+	"context"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+)
+
+// Batch handle requests sent in batch
+func (endpoint *Endpoint) Batch(ctx context.Context, req *pb.BatchRequest) (resp *pb.BatchResponse, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	resp = &pb.BatchResponse{}
+
+	resp.Responses = make([]*pb.BatchResponseItem, 0, len(req.Requests))
+
+	// TODO find a way to pass some parameters between request -> response > request
+	// TODO maybe use reflection to shrink code
+	for _, request := range req.Requests {
```

Comment:
would it make sense to do some actions concurrently ?

Reply:
maybe at some point. We would need logic to determine what we can do in parallel. I think it's out of scope for now.